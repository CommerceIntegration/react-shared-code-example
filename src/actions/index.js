export { getPeople, getNextPeople, getPreviousPeople } from './people';
