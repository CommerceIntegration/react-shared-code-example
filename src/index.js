export * from './actions';
export { applySwitch } from './components/apply-switch';
export { default as reducers } from './reducers';
