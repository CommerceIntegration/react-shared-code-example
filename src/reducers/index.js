import people from './people';

/**
 * Exports a map of all reducers. In a real project this map would obviously be larger, but is
 * alright for example purposes.
 *
 * @export
 * @default
 */
export default {
  people,
};
