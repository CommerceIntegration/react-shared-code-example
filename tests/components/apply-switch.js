import React from 'react';
import renderer from 'react-test-renderer';

import { applySwitch } from '../../src';

const Full = () => (
  <div>
    <p>Full</p>
  </div>
);

const Loading = () => (
  <div>
    <p>Loading</p>
  </div>
);

describe('components', () => {
  describe('apply-switch', () => {
    it('matches its snapshot and calls loadInitial if not fullRender', () => {
      const Skeleton = applySwitch(Full, Loading);
      const loadInitial = jest.fn();

      const tree = renderer
        .create(<Skeleton fullRender={false} loadInitial={loadInitial} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
      expect(loadInitial).toHaveBeenCalledTimes(1);
    });

    it('matches its snapshot and doesn\'t call loadInitial if fullRender', () => {
      const Skeleton = applySwitch(Full, Loading);
      const loadInitial = jest.fn();

      const tree = renderer
        .create(<Skeleton fullRender loadInitial={loadInitial} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
      expect(loadInitial).toHaveBeenCalledTimes(0);
    });
  });
});
