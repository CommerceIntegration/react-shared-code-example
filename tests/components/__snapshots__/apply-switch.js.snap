// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components apply-switch matches its snapshot and calls loadInitial if not fullRender 1`] = `
<div>
  <p>
    Loading
  </p>
</div>
`;

exports[`components apply-switch matches its snapshot and doesn't call loadInitial if fullRender 1`] = `
<div>
  <p>
    Full
  </p>
</div>
`;
